src/joystick/hidapi/SDL_hidapi_switch2.c

```diff
@@ -0,0 +1,266 @@
+        if (size > 14) {
+            Uint64 timestamp = SDL_GetTicksNS();
+            for (size_t i = 0; i < SDL_arraysize(buttons); ++i) {
+                SDL_SendJoystickButton(
+                    timestamp,
+                    joystick,
+                    (Uint8) i,
+                    (packet[buttons[i].byte] & buttons[i].mask) != 0
+                );
+            }
+            SDL_SendJoystickAxis(
+                timestamp,
+                joystick,
+                SDL_GAMEPAD_AXIS_LEFTX,
+                (Sint16) HIDAPI_RemapVal(
+                    (float) (packet[6] | ((packet[7] & 0x0F) << 8)),
+                    0,
+                    4096,
+                    SDL_MIN_SINT16,
+                    SDL_MAX_SINT16
+                )
+            );
+            SDL_SendJoystickAxis(
+                timestamp,
+                joystick,
+                SDL_GAMEPAD_AXIS_LEFTY,
+                (Sint16) HIDAPI_RemapVal(
+                    (float) ((packet[7] >> 4) | (packet[8] << 4)),
+                    0,
+                    4096,
+                    SDL_MIN_SINT16,
+                    SDL_MAX_SINT16
+                )
+            );
+            SDL_SendJoystickAxis(
+                timestamp,
+                joystick,
+                SDL_GAMEPAD_AXIS_RIGHTX,
+                (Sint16) HIDAPI_RemapVal(
+                    (float) (packet[9] | ((packet[10] & 0x0F) << 8)),
+                    0,
+                    4096,
+                    SDL_MIN_SINT16,
+                    SDL_MAX_SINT16
+                )
+            );
+            SDL_SendJoystickAxis(
+                timestamp,
+                joystick,
+                SDL_GAMEPAD_AXIS_RIGHTY,
+                (Sint16) HIDAPI_RemapVal(
+                    (float) ((packet[10] >> 4) | (packet[11] << 4)),
+                    0,
+                    4096,
+                    SDL_MIN_SINT16,
+                    SDL_MAX_SINT16
+                )
+            );
+            SDL_SendJoystickAxis(
+                timestamp,
+                joystick,
+                SDL_GAMEPAD_AXIS_LEFT_TRIGGER,
+                (Sint16) HIDAPI_RemapVal(packet[13], 0, 255, SDL_MIN_SINT16, SDL_MAX_SINT16)
+            );
+            SDL_SendJoystickAxis(
+                timestamp,
+                joystick,
+                SDL_GAMEPAD_AXIS_RIGHT_TRIGGER,
+                (Sint16) HIDAPI_RemapVal(packet[14], 0, 255, SDL_MIN_SINT16, SDL_MAX_SINT16)
+            );
+        }
+    }
+    return true;
+}
+
+static bool HIDAPI_DriverSwitch2_OpenJoystick(SDL_HIDAPI_Device *device, SDL_Joystick *joystick)
+{
+    // Initialize the joystick capabilities
+    joystick->nbuttons = 21;
+    joystick->naxes = SDL_GAMEPAD_AXIS_COUNT;
+
+    return true;
+}
+
+static bool HIDAPI_DriverSwitch2_RumbleJoystick(SDL_HIDAPI_Device *device, SDL_Joystick *joystick, Uint16 low_frequency_rumble, Uint16 high_frequency_rumble)
+{
+    return false;
```

Comment:
These functions need to set an error. Maybe this instead?
```suggestion
    return SDL_Unsupported();
```

Reply:
Did this for all `return false` paths that weren't support queries.